### PR TITLE
Compressor fixes and improvements

### PR DESCRIPTION
**Pull request by gentoo90**
New package structure:
```
packages
\-- RequireJsNet.Compressor.2.1.37
    +-- build
    │   \-- RequireJsNet.Compressor.targets
    \-- tools
        +-- EcmaScript.NET.dll
        +-- Jint.dll
        +-- Microsoft.Web.Infrastructure.dll
        +-- Newtonsoft.Json.dll
        +-- RequireJsNet.Compressor.dll
        +-- RequireJsNet.Compressor.pdb
        +-- RequireJsNet.dll
        +-- RequireJsNet.pdb
        +-- System.Web.Mvc.dll
        +-- System.Web.Razor.dll
        +-- System.Web.WebPages.Deployment.dll
        +-- System.Web.WebPages.Razor.dll
        +-- System.Web.WebPages.dll
        +-- Yahoo.Yui.Compressor.dll
        \-- YuiCompressor.NET.license.txt
```

* `RequireJsNet.Compressor.targets` automatically included into the project.
* No `dll`s are added to the project itself.
* All dependencies are packed and with correct versions.

In simple case user only has to install the package and create `ReqiureJS.json` and bundles will compile. If other config files are needed they can be added by setting `BuildAction` to `RequireJsNetConfig`:
requirejsnetbuildaction

Inspired by [RazorGenerator.MsBuild](https://github.com/RazorGenerator/RazorGenerator/wiki/Using-RazorGenerator.MsBuild) and [LessMsbuildTasks](https://github.com/Jehran/LessMsbuildTasks)

### CODE REVIEW

**Comment by gentoo90** [comment]
CC #71

**Comment by stefanprodan** [comment]
This crashes the TeamCity NuGet publish job:

```
[16:42:26]Step 1/3: MSBuild (11s)
[16:42:30][Step 1/3] RequireJsNet\RequireJsNet.csproj.teamcity: Build target: Build (6s)
[16:42:37][RequireJsNet\RequireJsNet.csproj.teamcity] BuildPackage
[16:42:37][BuildPackage] Exec
[16:42:37][Exec] C:\BuildAgent1\work\1f08d78398aef0cc\.nuget\NuGet.targets(100, 9): Please specify a nuspec or project file to use.
[16:42:37][Exec] C:\BuildAgent1\work\1f08d78398aef0cc\.nuget\NuGet.targets(100, 9): error MSB3073: The command ""..\.nuget\NuGet.exe" pack "C:\BuildAgent1\work\1f08d78398aef0cc\RequireJsNet\RequireJsNet.csproj.teamcity" -Properties Configuration=Debug -NonInteractive  -OutputDirectory "C:\BuildAgent1\work\1f08d78398aef0cc\RequireJsNet\bin\Debug" -symbols" exited with code 1.
[16:42:38][Step 1/3] Step MSBuild failed
```

**Comment by gentoo90** [comment]
> `pack "C:\BuildAgent1\work\1f08d78398aef0cc\RequireJsNet\RequireJsNet.csproj.teamcity"`

Looks like project name should match nuspec name

File: .nuget/NuGet.targets [hunk 1]
```diff
@@ -7,7 +7,7 @@
         <RestorePackages Condition="  '$(RestorePackages)' == '' ">false</RestorePackages>
 
         <!-- Property that enables building a package from a project -->
-        <BuildPackage Condition=" '$(BuildPackage)' == '' ">false</BuildPackage>
+        <BuildPackage Condition=" '$(BuildPackage)' == '' ">true</BuildPackage>
```
Comment (gentoo90) [inline .nuget/NuGet.targets]: Tried to run nuget manually and it doesn't like `.csproj` with `.teamcity` extension.
I enabled nuget build here. So it probably should be disabled for teamcity.

**Comment by gentoo90** [comment]
Also you'll probably need to check "Create **tool** package" checkbox
nuget6

**Comment by stefanprodan** [comment]
You can see my TeamCity config using guest account here http://tcoss.veritech.io/project.html?projectId=RequireJSNet

**Comment by stefanprodan** [comment]
I'm updating to latest VS.NET on the build server so I can switch to 2015 build tools, restarting server now. Create tool package doesn't fix the build.

**Comment by stefanprodan** [comment]
Switching to MS Build 2015 didn't fixed the build.

**Comment by stefanprodan** [comment]
I've also update the TeamCity NuGet to 3.4.3 but still crashes...

**Comment by gentoo90** [comment]
I guess I've found it. Set `BuildPackage` to `false` in TeamCity build parameters:
teamcityparameter
It will be added to msbuild properties and disable additional `nuget pack` runned by msbuild itself. After that TeamCity nuget step will proceed.

In `csproj` `BuildPackage` is set to `true` so users are able to build nuget package in Visual Studio.

**Comment by stefanprodan** [comment]
Both packages are on NuGet. Thank you very much!

**Pull request merged**